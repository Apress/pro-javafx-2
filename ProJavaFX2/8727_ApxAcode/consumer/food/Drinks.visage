/*
 * @subtest
 */

// Drinks.visage
package food;

public class Coffee {
  public var brand: String;
}

public class Tea {
  public var kind: String;
}
package var drinksOffered:Integer;

public function getCoffee():Coffee {
  drinksOffered++;
  Coffee { brand: "Folgers" }
}
public function getTea():Tea {
  drinksOffered++;
  Tea { kind: "Iced" }
}
public function numberOfDrinksOffered() {
  drinksOffered
}
