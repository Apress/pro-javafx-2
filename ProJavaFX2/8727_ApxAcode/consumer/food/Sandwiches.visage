/*
 * @subtest
 */

// Sandwiches.visage
package food;

public class Club {
  public-init var kind: String;
  public-read var price: Number;
}

public function calculatePrice(club: Club) {
  if (club.kind == "Roast Beef") {
    club.price = 7.99;
  } else if (club.kind == "Chicken") {
    club.price = 6.99;
  }
}
