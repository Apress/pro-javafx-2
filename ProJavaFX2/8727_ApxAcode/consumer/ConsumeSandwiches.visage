/*
 * @test
 * @compile ../food/Sandwiches.visage
 * @package consumer
 * @run
 */

package consumer;

import food.Sandwiches.*;

var club = Club { kind: "Roast Beef" };
calculatePrice(club);
println("The price of the {club.kind} club sandwich is {club.price}.");
