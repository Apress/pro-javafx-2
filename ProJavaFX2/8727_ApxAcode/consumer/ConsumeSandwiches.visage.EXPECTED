The price of the Roast Beef club sandwich is 7.99.
